src/map/script.cpp

```diff
@@ -11337,18 +11337,28 @@ BUILDIN_FUNC(sc_end)
 
 /**
  * Ends all status effects from any learned skill on the attached player.
- * sc_end_class {<char_id>};
+ * if <job_id> was given it will end all the effect for that job
+ * if <job_id> is 0 it will return the job of the invoking character
+ * sc_end_class {<job_id>{,<char_id>}};
  */
 BUILDIN_FUNC(sc_end_class)
 {
 	struct map_session_data *sd;
 	uint16 skill_id;
-	int i;
+	int class_ = 0;
+	struct block_list *bl = map_id2bl(st->rid);
 
-	if (!script_charid2sd(2, sd))
-		return SCRIPT_CMD_FAILURE;
+	if (!script_charid2sd(3, sd)) {
+		script_pushint(st, -1);
+		return SCRIPT_CMD_SUCCESS;
+	}
+
+	if (script_hasdata(st, 2))
+		class_ = script_getnum(st, 2);
```

Comment:
probably adding a class validation into this if would be nice, otherwise you might fail in the skill tree lookup

Reply:
i will search about it << still don't know a lot of things

Reply:
just to make sure
you want something like
`if (class_ != pc_jobid2mapid(class_))`
`			return SCRIPT_CMD_FAILURE;`
so the map server wont crash if the class id was not a class id right ?

Reply:
pcdb_checkid